staging/src/k8s.io/apimachinery/pkg/runtime/serializer/cbor/internal/modes/decode_test.go

```diff
@@ -275,7 +282,287 @@ func TestDecode(t *testing.T) {
 			},
 			assertOnError: assertNilError,
 		},
-	} {
+		{
+			name:          "int64 minimum positive value",
+			in:            hex("00"), // 0
+			want:          int64(0),
+			assertOnError: assertNilError,
+		},
+		{
+			name:          "int64 max positive value",
+			in:            hex("1b7fffffffffffffff"), // 9223372036854775807
+			want:          int64(9223372036854775807),
+			assertOnError: assertNilError,
+		},
+		{
+			name:          "int64 max negative value",
+			in:            hex("20"), // -1
+			want:          int64(-1),
+			assertOnError: assertNilError,
+		},
+		{
+			name:          "int64 min negative value",
+			in:            hex("3b7fffffffffffffff"), // -9223372036854775808
+			want:          int64(-9223372036854775808),
+			assertOnError: assertNilError,
+		},
+		{
+			name:          "empty byte string",
+			in:            hex("40"), // ''
+			want:          "",
+			assertOnError: assertNilError,
+		},
+		{
+			name:          "empty text string",
+			in:            hex("60"), // ""
+			want:          "",
+			assertOnError: assertNilError,
+		},
+		{
+			name: "map with different key types",
+			in:   hex("a201616161626132"), // { 1: "a", "b": "2"}
+			assertOnError: assertOnConcreteError(func(t *testing.T, e *cbor.UnmarshalTypeError) {
+				if e.CBORType != "positive integer" || e.GoType != "string" {
+					t.Errorf("Expected types: positive integer,string. Got: %#v,%#v", e.CBORType, e.GoType)
+				}
+			}),
+		},
+		{
+			name: "map with non string key types",
+			in:   hex("a1fb40091eb851eb851f63706965"), // { 3.14: "pie"}
+			assertOnError: assertOnConcreteError(func(t *testing.T, e *cbor.UnmarshalTypeError) {
+				if e.CBORType != "primitives" || e.GoType != "string" {
+					t.Errorf("Expected types: primitives,string. Got: %#v,%#v", e.CBORType, e.GoType)
+				}
+			}),
+		},
+		{
+			name:          "map with byte string key type",
+			in:            hex("a143abcdef187b"), // {h'ABCDEF': 123}
+			want:          map[string]interface{}{"\xab\xcd\xef": int64(123)},
+			assertOnError: assertNilError,
+		},
+		{
+			name:  "max positive integer value supported by cbor: 2^64 - 1",
+			modes: []cbor.DecMode{modes.Decode},
+			in:    hex("1bffffffffffffffff"), // 18446744073709551615
+			assertOnError: assertOnConcreteError(func(t *testing.T, e *cbor.UnmarshalTypeError) {
+				if e.CBORType != "positive integer" || e.GoType != "int64" {
+					t.Errorf("Expected types: positive integer,int64. Got: %#v,%#v", e.CBORType, e.GoType)
```

Comment:
You may need to assert on the result of `Error() string` here, since that's the only way we can tell that the error is due to overflow.

Reply:
Fixed.